http/server.ts

```diff
@@ -371,21 +374,32 @@ export class Server {
           error instanceof Deno.errors.ConnectionReset ||
           error instanceof Deno.errors.NotConnected
         ) {
-          // Backoff after transient errors to allow time for the system to
-          // recover, and avoid blocking up the event loop with a continuously
-          // running loop.
-          if (!acceptBackoffDelay) {
-            acceptBackoffDelay = INITIAL_ACCEPT_BACKOFF_DELAY;
-          } else {
-            acceptBackoffDelay *= 2;
-          }
-
-          if (acceptBackoffDelay >= MAX_ACCEPT_BACKOFF_DELAY) {
-            acceptBackoffDelay = MAX_ACCEPT_BACKOFF_DELAY;
+          if (!this.#closed) {
+            // Backoff after transient errors to allow time for the system to
+            // recover, and avoid blocking up the event loop with a continuously
+            // running loop.
+            if (!acceptBackoffDelay) {
+              acceptBackoffDelay = INITIAL_ACCEPT_BACKOFF_DELAY;
+            } else {
+              acceptBackoffDelay *= 2;
+            }
+
+            if (acceptBackoffDelay >= MAX_ACCEPT_BACKOFF_DELAY) {
+              acceptBackoffDelay = MAX_ACCEPT_BACKOFF_DELAY;
+            }
+
+            await delay(acceptBackoffDelay, {
+              signal: this.#acceptBackoffDelayAbortController.signal,
+            }).catch((err: unknown) => {
```

Comment:
Can you avoid mixing `async/await` with `Promise` APIs. In other words, make this a `try...catch`.

Reply:
Yes, that's reasonable. I made the change here: https://github.com/denoland/deno_std/pull/2732/commits/4f7ef9c987c4b3feba1b82649ccce3b646a82f5c